### DISCUSSION TITLE
Wrong values set_lreal

### DISCUSSION THREAD

**Discussion by Joren-Vdbosch** [gijzelaerr/python-snap7#484] [Q&A] [answered]
When i use set_lreal to write values into a database all the data is the same:
If the value that i want to put in a database is bigger than 1 it will be 2.
If the original value is smaller than 1 it will be 3.0517578125E-05 in the database.

What can i do to fix it?

**Comment by spreeker** [comment]
I could be that not all util functions have been properly tested and used in production.  The code is quite sort and easy to understand. Mayby you can make a fix in:     

` struct.pack_into(">d", bytearray_, byte_index, lreal)`

https://github.com/gijzelaerr/python-snap7/blob/5f9de736fb379f984c5211cf65759cd7687706c1/snap7/util.py#L1070

Could aslo by that the db has a prefix so all byte values of moved a bit.

**Comment by gijzelaerr** [comment, marked as answer]
Closing as likely resolved. The `set_lreal` bug was fixed in PR #444 (same bug reported in #427). Please upgrade to the latest version.
